agents/matlab/matlab_agent/src/core/interactive.py

```diff
@@ -0,0 +1,302 @@
+        self.host = host
+        self.port = port
+        self._srv: Optional[socket.socket] = None
+        self._conn: Optional[socket.socket] = None
+        self._buffer = b""
+        self.matlab_proc: Optional[subprocess.Popen] = None
+
+    def start(self) -> None:
+        self._srv = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self._srv.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self._srv.bind((self.host, self.port))
+        self._srv.listen()
+
+    def accept(self) -> None:
+        if not self._srv:
+            raise RuntimeError("Server not started")
+        ready = select([self._srv], [], [], 60)
+        if ready[0]:
+            self._conn, _ = self._srv.accept()
+            self._conn.setblocking(False)
+        else:
+            logger.error("[INTERACTIVE] Timeout waiting for client connection.")
+            raise TimeoutError("No client connection received in time.")
+
+    def send(self, data: Dict[str, Any]) -> None:
+        if self._conn:
+            payload = json.dumps(data).encode() + b"\n"
+            self._conn.sendall(payload)
+
+    def recv_all(self) -> list[Dict[str, Any]]:
+        if not self._conn or not select([self._conn], [], [], 0)[0]:
+            return []
+
+        self._buffer += self._conn.recv(4096)
+        lines = self._buffer.split(b"\n")
+        self._buffer = lines[-1]
+        messages: list[Dict[str, Any]] = []
+        for line in lines[:-1]:
+            line = line.strip()
+            if not line:
+                continue
+            try:
+                messages.append(json.loads(line.decode()))
+            except json.JSONDecodeError as exc:  # pragma: no cover - logging only
+                logger.error("[INTERACTIVE] Invalid JSON: %s", exc)
+        return messages
+
+    def close(self) -> None:
+        if self._conn:
+            self._conn.close()
+        if self._srv:
+            self._srv.close()
+        if self.matlab_proc and self.matlab_proc.poll() is None:
+            self.matlab_proc.terminate()
+            try:
+                self.matlab_proc.wait(timeout=5)
+            except subprocess.TimeoutExpired:  # pragma: no cover - best effort
+                self.matlab_proc.kill()
+
+
+# ---------------------------------------------------------------------------
+# Controller
+# ---------------------------------------------------------------------------
+
+class MatlabInteractiveController:
+    def __init__(
+        self,
+        path: str,
+        file: str,
+        source: str,
+        broker: IMessageBroker,
+        templates: Dict[str, Any],
+        tcp_cfg: Dict[str, Any],
+        bridge_meta: str,
+        request_id: str,
+        agent_id: str = "agent",
+    ) -> None:
+        self.sim_path = Path(path).resolve()
+        self.sim_file = file
+        if not (self.sim_path / self.sim_file).exists():
+            raise FileNotFoundError(self.sim_file)
+
+        self.source = source
```

Comment:
This kind of mapping is common across batch, streaming and interactive modes. Can all the header processing and augumntation be refactored into `metadata.py`?